cinema/permissions.py

```diff
@@ -0,0 +1,38 @@
+import logging
+
+from rest_framework.permissions import BasePermission, SAFE_METHODS
+
+logger = logging.getLogger(__name__)
+
+
+class IsAdminOrIfAuthenticatedReadOnly(BasePermission):
+    """
+    Allows read/write access for admin users (except delete),
+    and read-only access for authenticated users.
+    """
+    def has_permission(self, request, view):
+        try:
+            if (
+                    view.action == "list" or (
+                        view.action == "retrieve"
+                        and view.__class__.__name__ in (
+                            "MovieViewSet",
+                            "MovieSessionViewSet"
+                        )
+                    )
+            ):
+                return request.user and request.user.is_authenticated

```

Comment:
The current logic allows authenticated users to perform read-only operations on 'list' and 'retrieve' actions for 'MovieViewSet' and 'MovieSessionViewSet'. Ensure this aligns with the task requirements, as it might need to be expanded to other viewsets as well.